internal/settings/settings_test.go

```diff
@@ -381,17 +354,22 @@ func TestSetDefaultFeatureGatesRespectsOverrides(t *testing.T) {
 	}
 }
 
-func TestCheckRuntimeParams_MemTotalLimitAndBallastEnvs(t *testing.T) {
+func TestSetSoftMemLimitWithoutGoMemLimitEnvVar(t *testing.T) {
+
+	// if GOLIMIT is not set, we expect soft limit to be 90% of the total memory env var or 90% of default total memory  512 Mib.
 	t.Cleanup(setRequiredEnvVars(t))
 	require.NoError(t, os.Setenv(MemTotalEnvVar, "200"))
-	require.NoError(t, os.Setenv(MemLimitMiBEnvVar, "150"))
-	require.NoError(t, os.Setenv(BallastEnvVar, "50"))
-
 	settings, err := New([]string{})
 	require.NoError(t, err)
 	require.NotNil(t, settings)
-	require.Equal(t, "50", os.Getenv(BallastEnvVar))
-	require.Equal(t, "150", os.Getenv(MemLimitMiBEnvVar))
+	require.Equal(t, int64(188743680), debug.SetMemoryLimit(100))
```

Comment:
```suggestion
	require.Equal(t, int64(188743680), debug.SetMemoryLimit(-1))
```
If I am following the test correctly, it seems we just want to retrieve the currently set limit. Same for the test below.

Reply:
The way this test work bit differently, The idea of setting setting SetMemoryLimit is not to match the set memory limit ( 100 )  rather each time you use that API it returns the previously set memorylimit if it had been set.  Test just asserts if it was the soft limit was set or not. For more information -> https://pkg.go.dev/runtime/debug#SetMemoryLimit

Reply:
yes, `SetMemoryLimit returns the previously set memory limit. A negative input does not adjust the limit, and allows for retrieval of the currently set memory limit.` seems like you just want to retrieve it and passing -1 will do just that

Reply:
That works too!